doc/guides/test/sanitycheck.rst

```diff
@@ -303,7 +303,16 @@ arch_exclude: <list of arches, such as x86, arm, arc>
     Set of architectures that this test case should not run on.
 
 platform_whitelist: <list of platforms>
-    Set of platforms that this test case should only be run for.
+    Set of platforms that this test case should only be run for. Do not use
+    this option to limit testing or building in CI due to time or resource
+    constraints, this option should only be used of the test or sample can
+    only be run on the whitelisted platform and nothing else.
+
+integration: <YML list of platforms/boards>
```

Comment:
Maybe `platform_integration_list` or `integration_platforms` instead of `integration`?

Reply:
Another proposal: drop all of this and do it completely differently...

basically the option would be --mode with a list of modes that are supported, for example:
``--mode [daily|pull-request|default|...]```


now, in the testcase yaml file this can be triggered with:

```

 test:
   mode:
     pull-request:
       - <platform>
       - <platform>
     daily:
       - all
```
Still need to figure out the syntax and allow flexibility here.

Existing `build_on_all` would fold into that and removed as a standalone option for example.

Reply:
ok, sticking with original proposal with a name change, now using integration_platforms.